src/components/search-by-message/SearchByMessage.tsx

```diff
@@ -1,31 +1,62 @@
-import { useState, ChangeEvent, FC } from 'react'
+import { useState, ChangeEvent, useEffect, FC } from 'react'
 import { useTranslation } from 'react-i18next'
 import Box from '@mui/material/Box'
 
 import InputWithIcon from '~/components/input-with-icon/InputWithIcon'
 import IconsWithCounter from '~/components/icons-with-counter/IconsWithCounter'
+import { useDebounce } from '~/hooks/use-debounce'
 
 import { styles } from '~/components/search-by-message/SearchByMessage.styles'
 
 interface SearchByMessageProps {
-  maxValue: number
+  messages: { text: string }[]
+  onFilteredMessagesChange: (filteredMessages: string[]) => void
+  onFilteredIndexChange: (filteredIndex: number) => void
 }
 
-const SearchByMessage: FC<SearchByMessageProps> = ({ maxValue = 10 }) => {
+const SearchByMessage: FC<SearchByMessageProps> = ({
+  messages,
+  onFilteredMessagesChange,
+  onFilteredIndexChange
+}) => {
   const { t } = useTranslation()
   const [search, setSearch] = useState<string>('')
+  const [findMessage, setFindMessage] = useState<string[]>([])
 
+  const debouncedOnFilteredMessagesChange = useDebounce(
+    (filteredMessages: string[]) => {
+      onFilteredMessagesChange(filteredMessages)
+    },
+    1500
```

Comment:
I guess this delay is too long, in my opinion, 500 will be enought.

Reply:
yep. i think the same, right now its .5 sec